File: dotnet/bindings/CLRInterface/TaskId.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Automatak.DNP3.Interface
{
    public enum TaskIds : int
    {
        CLEAR_RESTART,
        DISABLE_UNSOLICITED,
        ASSIGN_CLASS,
        STARTUP_INTEGRITY_POLL,
        SERIAL_TIME_SYNC,
        ENABLE_UNSOLICITED,
        EVENT_SCAN
    };

    public class TaskId
    {
        public TaskId(int id, bool isUserAssigned)
        {
            this.id = id;
            this.isUserAssigned = isUserAssigned;
        }

        public int Id
        {
            get
            {
                return id;
            }
        }

        public bool IsUserAssigned
        {
            get
            {
                return isUserAssigned;
            }
        }

        readonly int id;
		readonly bool isUserAssigned;

		
    }
}

File: generation/dnp3/src/main/scala/com/automatak/render/dnp3/enums/TaskId.scala
package com.automatak.render.dnp3.enums

import com.automatak.render._


object TaskId {

  private val comments = List("Enumeration of internal tasks")

  def apply(): EnumModel = EnumModel("TaskId", comments, EnumModel.Integer, codes, Base10)

  private val codes = List(
      EnumValue("CLEAR_RESTART", 0),
      EnumValue("DISABLE_UNSOLICITED", 1),
      EnumValue("ASSIGN_CLASS", 2),
      EnumValue("STARTUP_INTEGRITY_POLL", 3),
      EnumValue("SERIAL_TIME_SYNC", 4),
      EnumValue("ENABLE_UNSOLICITED", 5),
      EnumValue("AUTO_EVENT_SCAN", 6),
      EnumValue("USER_TASK", 7)
  )

}



